Unable to install wrangler

### 🐛 Bug report

I'm unable to install wrangler. Most probably reason is the binary-install stuff in 1.14

#### Describe the bug

I try to install wrangler. It fails :D

```
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/@cloudflare/wrangler
npm ERR! command failed
npm ERR! command sh -c node ./install-wrangler.js
npm ERR! internal/modules/cjs/loader.js:883
npm ERR!   throw err;
npm ERR!   ^
npm ERR!
npm ERR! Error: Cannot find module '@cloudflare/binary-install'
npm ERR! Require stack:
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/binary.js
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js
npm ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19)
npm ERR!     at require (internal/modules/cjs/helpers.js:88:18)
npm ERR!     at Object.<anonymous> (/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js:1:20)
npm ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
npm ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
npm ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19) {
npm ERR!   code: 'MODULE_NOT_FOUND',
npm ERR!   requireStack: [
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js',
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js'
npm ERR!   ]
npm ERR! }
```
#### Reproduce the bug

`npm i -g @cloudflare/wrangler`

#### Expected behavior

It installs wrangler

#### Environment and versions

Fill out the following information about your environment.

* operating system: macOS Big Sur 11.1
* output of `wrangler -V`: 👷 ✨  wrangler 1.13.0
* output of `node -v`: v14.16.0
* content of `wrangler.toml`: N/A

Yeah, it's back when I do that.

But isn't the solution to just set binary-install as a dependency to wrangler?

```
$ rm -rf ~/.wrangler
$ npm i -g @cloudflare/wrangler
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/@cloudflare/wrangler
npm ERR! command failed
npm ERR! command sh -c node ./install-wrangler.js
npm ERR! internal/modules/cjs/loader.js:883
npm ERR!   throw err;
npm ERR!   ^
npm ERR!
npm ERR! Error: Cannot find module '@cloudflare/binary-install'
npm ERR! Require stack:
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/binary.js
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js
npm ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19)
npm ERR!     at require (internal/modules/cjs/helpers.js:88:18)
npm ERR!     at Object.<anonymous> (/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js:1:20)
npm ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
npm ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
npm ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19) {
npm ERR!   code: 'MODULE_NOT_FOUND',
npm ERR!   requireStack: [
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js',
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js'
npm ERR!   ]
npm ERR! }
```

Looking at the verbose logs, doesn't it look like it's running postinstall before it's running install for binary?
```
npm i -g --verbose @cloudflare/wrangler                                      ✔  10196  15:22:35
npm verb cli [
npm verb cli   '/usr/local/bin/node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'i',
npm verb cli   '-g',
npm verb cli   '--verbose',
npm verb cli   '@cloudflare/wrangler'
npm verb cli ]
npm info using npm@7.6.0
npm info using node@v14.16.0
npm timing config:load:defaults Completed in 1ms
npm timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 1ms
npm timing config:load:builtin Completed in 1ms
npm timing config:load:cli Completed in 1ms
npm timing config:load:env Completed in 1ms
npm timing config:load:file:/Users/oskar.risberg/repos/test/.npmrc Completed in 0ms
npm timing config:load:project Completed in 0ms
npm timing config:load:file:/Users/oskar.risberg/.npmrc Completed in 1ms
npm timing config:load:user Completed in 1ms
npm timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
npm timing config:load:global Completed in 0ms
npm timing config:load:cafile Completed in 0ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:setUserAgent Completed in 1ms
npm timing config:load:setEnvs Completed in 0ms
npm timing config:load Completed in 6ms
npm verb npm-session d1c49cc91c6a9001
npm timing npm:load Completed in 15ms
npm timing arborist:ctor Completed in 1ms
npm timing idealTree:init Completed in 6ms
npm timing idealTree:userRequests Completed in 2ms
npm http fetch GET 304 https://registry.npmjs.org/@cloudflare%2fwrangler 304ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/@cloudflare%2fbinary-install 224ms (from cache)
npm timing idealTree:#root Completed in 539ms
npm timing idealTree:buildDeps Completed in 541ms
npm timing idealTree:fixDepFlags Completed in 1ms
npm timing idealTree Completed in 551ms
npm timing reify:loadTrees Completed in 575ms
npm timing reify:diffTrees Completed in 3ms
npm timing reify:retireShallow Completed in 4ms
npm timing reify:createSparse Completed in 0ms
npm timing reify:loadBundles Completed in 0ms
npm timing reifyNode:node_modules/@cloudflare/wrangler Completed in 19ms
npm timing reify:unpack Completed in 19ms
npm timing reify:unretire Completed in 1ms
npm timing build:queue Completed in 1ms
npm timing build:link:node_modules/@cloudflare/wrangler Completed in 2ms
npm timing build:link Completed in 2ms
npm info run @cloudflare/wrangler@1.14.0 postinstall node_modules/@cloudflare/wrangler node ./install-wrangler.js
npm info run @cloudflare/wrangler@1.14.0 postinstall { code: 1, signal: null }
npm timing reify:rollback:createSparse Completed in 3ms
npm timing reify:rollback:retireShallow Completed in 1ms
npm timing command:install Completed in 702ms
npm verb stack Error: command failed
npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
npm verb stack     at ChildProcess.emit (events.js:315:20)
npm verb stack     at maybeClose (internal/child_process.js:1048:16)
npm verb stack     at Socket.<anonymous> (internal/child_process.js:439:11)
npm verb stack     at Socket.emit (events.js:315:20)
npm verb stack     at Pipe.<anonymous> (net.js:673:12)
npm verb pkgid @cloudflare/wrangler@1.14.0
npm verb cwd /Users/oskar.risberg/repos/test
npm verb Darwin 20.2.0
npm verb argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "-g" "--verbose" "@cloudflare/wrangler"
npm verb node v14.16.0
npm verb npm  v7.6.0
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/@cloudflare/wrangler
npm ERR! command failed
npm ERR! command sh -c node ./install-wrangler.js
npm ERR! internal/modules/cjs/loader.js:883
npm ERR!   throw err;
npm ERR!   ^
npm ERR!
npm ERR! Error: Cannot find module '@cloudflare/binary-install'
npm ERR! Require stack:
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/binary.js
npm ERR! - /usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js
npm ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19)
npm ERR!     at require (internal/modules/cjs/helpers.js:88:18)
npm ERR!     at Object.<anonymous> (/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js:1:20)
npm ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
npm ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
npm ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
npm ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
npm ERR!     at Module.require (internal/modules/cjs/loader.js:952:19) {
npm ERR!   code: 'MODULE_NOT_FOUND',
npm ERR!   requireStack: [
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/binary.js',
npm ERR!     '/usr/local/lib/node_modules/@cloudflare/wrangler/install-wrangler.js'
npm ERR!   ]
npm ERR! }
npm verb exit 1
npm timing npm Completed in 898ms
npm verb code 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,19 @@
 # Changelog
 
+## 1.14.1
+
+- ### Fixes
+
+  - **revert default install location change - [xortive], [pull/1798]**
+
+    In 1.14.0, we changed the default install location from `~/.wrangler` to `node_modules`,
+    to allow `npx wrangler` to use the pinned version in a project's dependencies. It seems that
+    this is causing issues, so we're rolling it back in favor of having folks who want this behavior,
+    to specify a the install location in the `config` section of package.json. We'll document this soon.
+
+    [xortive]: https://github.com/xortive
+    [pull/1798]: https://github.com/cloudflare/wrangler/pull/1798
+
 ## 1.14.0
 
 - ### Features

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "wrangler"
-version = "1.14.0"
+version = "1.14.1"
 authors = ["The Wrangler Team <wrangler@cloudflare.com>", "Avery Harnish <averyharnish@gmail.com>", "Ashley Lewis <ashleymichal@gmail.com>", "Ashley Williams <ashley666ashley@gmail.com>", "Nat Davidson <davidson@cloudflare.com>", "Steve Manuel <nilslice@gmail.com>"]
 edition = "2018"
 license = "MIT/Apache-2.0"

--- a/RELEASE_CHECKLIST.md
+++ b/RELEASE_CHECKLIST.md
@@ -43,14 +43,21 @@ Most of your comments will be about the changelog. Once the PR is finalized and 
 
 1. If you made changes, squash or fixup all changes into a single commit.
 1. Run `git push` and wait for CI to pass.
+## Merge
+
+1. Hit the big green Merge button on the release PR.
+1. `git checkout master` and `git pull --rebase origin master`
 
 ### Tag and build release
 
 This part of the release process is handled by GitHub Actions, and our binaries are distributed as GitHub Releases. When you push a version tag, it kicks off an action that creates a new GitHub release for that tag, builds release binaries and attaches them to the release.
 
-1. Once ready to merge, tag the commit by running either `git tag -a v#.#.# -m #.#.#` (release), or `git tag -a v#.#.#-rc.# -m #.#.#` (release candidate)
+1. After pulling `master` in the step above, tag the commit by running either `git tag -a v#.#.# -m #.#.#` (release), or `git tag -a v#.#.#-rc.# -m #.#.#` (release candidate)
 1. Run `git push --tags`.
 1. Wait for CI to pass.
+1. If CI fails, delete the tag locally and remotely
+1. Fix whatever caused the CI failure
+1. Re-tag the healthy commit, and wait for CI to pass again.
 
 ### Edit the release
 
@@ -73,11 +80,6 @@ After CI builds the release binaries and they appear on the [releases page](http
    ```
 
    The new release candidate should then include updated testing instructions with a small changelog at the top to get folks who installed the old release candidate up to speed.
-
-## Publish
-
-1. Hit the big green Merge button on the release PR.
-1. `git checkout master` and `git pull --rebase origin master`
 
 ### Publish to crates.io (full release only)
 

--- a/npm/binary.js
+++ b/npm/binary.js
@@ -38,7 +38,7 @@ const getBinary = () => {
   const customPath =
     process.env.WRANGLER_INSTALL_PATH ||
     process.env.npm_config_wrangler_install_path;
-  const installDirectory = join(customPath || __dirname, "wrangler");
+  const installDirectory = join(customPath || os.homedir(), ".wrangler");
   return new Binary(url, { name: "wrangler", installDirectory });
 };
 

--- a/npm/package.json
+++ b/npm/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@cloudflare/wrangler",
-  "version": "1.14.0",
+  "version": "1.14.1",
   "description": "Command-line interface for all things Cloudflare Workers",
   "main": "binary.js",
   "scripts": {

